### PR TITLE
Disables progtots

### PR DESCRIPTION
**Pull request by Zonespace27**
## About The Pull Request

Reverts traitors to 2 objectives + escape/martyr/hijack

## How This Contributes To The Skyrat Roleplay Experience
Progression traitors was a mistake that doesn't promote RP in most cases

## Changelog

:cl:
del: Removed Progression Traitors, traitors are now back to two objectives + escape/martyr/hijack
/:cl:

### CODE REVIEW

**Comment by theselfish** [comment]
>switching back to quasi-ambitions seems like a good experiment

Didn't the majority of people agree that ambitions were a horrible idea?

**Comment by Aether217** [comment]
> > switching back to quasi-ambitions seems like a good experiment
> 
> Didn't the majority of people agree that ambitions were a horrible idea?

No, the idea was to have prog traitors as something on the side for those who "can not come up with a gimmick" but the more prog traitor ran it's course the more it showed that it is not something that fits with Skyrat. Progtot is goood for servers with short rounds and less restrictive rules about RP and escalation as it encourages fast and loose playstyles.

**Comment by alin-but-real** [comment]
CAVEAT: if the opfor isnt approved in 15 minutes it should auto-approve. having your traitor round nulled because no admin came and approved it is dumb dumb

also, how the fuck do you do lethal traitors now? admins are ghonna go "nuh you can't kill this person without asking!" and not approve your opfor. progtot at least offered traitors total freedom without admin intervention

**Comment by Higgin** [comment]
an automatic approval timer is a good idea for the same reason it was during ambitions, where folks on staff have said one of the things nobody wanted to pick up was approving them 

this is generally a bad idea and will tend towards staff minimizing what gets complaints which often get made in the heat of the moment

either entirely remove rolled antagonism, give us normal traitor, or stick with progtot - ambitions was not a good system and led to a shitty security-throttled loop where the only time things could get loud were when security was present and it was directed at them, specifically

if people want to do gimmicks or opfor, you can actually just approve them, even if there are antags in the round already

**Comment by Gandalf2k15** [comment]
We had the solution to this before and it was requiring them to submit for TC, having it this way will likely just get people into trouble for finding loopholes and will massively decrease the quality of traitors as we seen with ambitions. 

I made that system to purposefully remove ambitions and its utterly disgraceful system of making people submit to simply do their role of being a traitor. Massive no no there, it defeats the whole purpose of the system. Adding more automatic approval systems will cause a MASSIVE headache for people on staff as it will approve bad opfors, and not having it will cause the players that have their opfors ignored to just stop playing antag. It happened before with ambitions and that is why I made OPFOR and that is why its a separate system. Don't punish people for rolling antagonist. It's very very bad taste and the echoes of ambitions run rife in that PR. So imma close it and you know the reasons, and we removed ambitions before for the same reasons. It will not attain your desired result, it will fuck traitors and changelings to the point no one wants to play them like we had before. 

And you're also counting on EVERYONE being creative which we know from prior experience is not the case with skyrat and that's okay. We won't force people to be creative.

Some traitor players NEED objectives to play otherwise they'll submit crap OPFORs because they have to, like we seen with the 10tc opfor approval thing. Good traitor players already use OPFOR. All you're doing is forcing the less creative ones, which ARE STILL ENTITLED TO PLAY ANTAG, onto a system that will cause them to get into trouble and bloat the already under checked opfor queue.

It will not only ruin my OPFOR system, but it will also ruin traitors for the few that used it. I remember what happened with ambitions.

**Comment by RatFromTheJungle** [comment]
"Some traitor players NEED objectives to play otherwise they'll submit crap OPFORs because they have to, like we seen with the 10tc opfor approval thing. Good traitor players already use OPFOR. All you're doing is forcing the less creative ones, which ARE STILL ENTITLED TO PLAY ANTAG, onto a system that will cause them to get into trouble and bloat the already under checked opfor queue."

antag isn't a right, the idea that it is, is what got us into this situation with terrible-quality antags powergaming, and forcing an arms-race between them and security before the rounds even started.

I remember what's happened since we've lost ambitions, antags getting categorically worse, less inspired, and all around less engaging to the point we're at a record low for people rolling them, it's almost exclusively the same four people, even before guncargo.

**Comment by Aether217** [comment]
> Some traitor players NEED objectives to play otherwise they'll submit crap OPFORs because they have to, like we seen with the 10tc opfor approval thing. Good traitor players already use OPFOR. All you're doing is forcing the less creative ones, which ARE STILL ENTITLED TO PLAY ANTAG, onto a system that will cause them to get into trouble and bloat the already under checked opfor queue.

Is a bad antagonist better than no antagonist? No, it is not and we have seen that it is not. I do not think we should catter to those who can't put a minimum of effort into their round and then be surprised when what we have is low quality antags that prefer mechanics over roleplay.

**Comment by hugmeorelse** [comment]
I had asked for this, and I was not updated that this PR was closed.

**Comment by Iamgoofball** [comment]
opfors function way better than ambitions did and have much better onboarding and UX, so keep that in mind when discussing this concept, we all remember how bad ambitions was after all

ultimately there's nothing wrong with progression traitor's objectives and concept, it's a well put together system, it just requires a player with nuance and the ability to string together a coherent and interesting story from the pieces handed to them, something that not everyone can do well

any concerns about underperformance from progression traitors or opfor is an admin issue at its core

**Comment by Zergspower** [comment]
God no, not auto-approvals. And progtot system doesnt fit SR's environment and rules, their objectives are bland and boring and add nothing to the shifts RP Scene. Oh i killed poly.. why? Oh I put a bug on a board inside a locked room.. why? I stole a blackbox that has zero functional use to the station and thus no one gives a shit.  Most if not all of their goals are done without any sort of RP involved or required, and the level of bad faith we've had to spend 30+ minutes on a ticket for is more than one which is at the point of ridiculous.   Progtots are fine on shorter round and less RP focused servers

**Comment by Iamgoofball** [comment]
If you can't at least come up with a half-assed explanation as to why you're doing acts of sabotage against the station, then you may need to learn improv fundamentals.

**Comment by Higgin** [comment]
Switching progtot back out for regular traitor seems like a fine idea to try to me, though it's a shame it'll mean stepping down from some of the cooler things progtot offered (i.e. the different objectives, final objectives.)

Suppse people should pool resources or roll contractor if they want the extra TC for spendier items now (not a bad thing imo.)

**Comment by Zonespace27** [comment]
> Switching progtot back out for regular traitor seems like a fine idea to try to me, though it's a shame it'll mean stepping down from some of the cooler things progtot offered (i.e. the different objectives, final objectives.)

Might re-add some of the more interesting progtot objectives later once this goes through, nothing guaranteed though

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
Instead of doing this, can you just add a new antagonist type that uses the old system? This way we can have both and it's more modular. Then you can update the dynamic midround to just spawn the new antag instead.

**Comment by Zonespace27** [comment]
> Instead of doing this, can you just add a new antagonist type that uses the old system? This way we can have both and it's more modular. Then you can update the dynamic midround to just spawn the new antag instead.

alright done

File: code/modules/antagonists/traitor/datum_traitor.dm
Comment (Gandalf2k15) [inline code/modules/antagonists/traitor/datum_traitor.dm]: reset this file etc

**Comment by ChungusGamer666** [comment]
> God no, not auto-approvals. And progtot system doesnt fit SR's environment and rules, their objectives are bland and boring and add nothing to the shifts RP Scene. Oh i killed poly.. why? Oh I put a bug on a board inside a locked room.. why? I stole a blackbox that has zero functional use to the station and thus no one gives a shit. Most if not all of their goals are done without any sort of RP involved or required, and the level of bad faith we've had to spend 30+ minutes on a ticket for is more than one which is at the point of ridiculous. Progtots are fine on shorter round and less RP focused servers

Give every objective a background explanation if that is such a big problem.

File: code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm
Comment (Zonespace27) [inline code/game/gamemodes/dynamic/dynamic_rulesets_latejoin.dm]:
```suggestion
	antag_datum = /datum/antagonist/traitor/saboteur //SKYRAT EDIT - PROGRESSIONLESS TRAITORS
```

File: modular_skyrat/master_files/code/modules/antagonists/traitor/datums/datum_traitor.dm
Comment (Zonespace27) [inline modular_skyrat/master_files/code/modules/antagonists/traitor/datums/datum_traitor.dm]:
```suggestion
	roundend_category = "Syndicate Saboteurs"
```

**Comment by Zenitheevee** [comment]
Alright
So, Instead of just mindlessly hitting the thumbs down button and submitting a long bitch of a post i kinda want to get some stuff out. 
<details>
  <summary>Concerns/Questions about Time/Rep locks</summary>
So, As we know in ProgTots, There are a lot of items that are reputation locked. This does play as a sort of adjustable artificial time lock for some of the items, and does require people to spend some time either doing objectives or, otherwise, laying low and earning stuff at a much slower rate. This does create a psuedo-balancing point by keeping certain items from being purchased from the get-go and going ham at the beginning of the round, And quite honestly having a look at some of the items, I dont think they're _quite_ balanced to openly exist without the limit. These include; Choosable loadouts that contain more than its cost in TC worth of gear or items that are _otherwise_ not available in the base traitor system (Dark Lord, for the biggest and most outstanding example), Roundstart *quite aggressive* weapons (desword, bulldog, C20), Syndicate modsuits (Normal having a required 200, with the elite having a hefty 900 requirement) and even things like POWERSINKS being available behind a fair amount of time requirements. 

So, The big question is how is _that_ being handled? Are the time locks still there just _now_ they're hardlocked behind the actual time (meaning things like the Elite modsuit that has a 900 time requirement being impossible to reach and even things like emags being unavailable for most if not hte whole round with its 200 requirement,), Or are they just available at the hat drop, Allowing someone to plant a power sink in the most unexplored parts of the station to make engineer's life a living hell? 

</details>

<details>
  <summary>TC Cost, spawn and balancing</summary>
So, This is another thing that kinda becomes an issue. Overlooking the OpFor system (Because its a lot more.. open and hands-on with moderators. One person could say you dont need 5tc to remove science from the ore silo, one person would suggest maybe going straight for the equipment you want rather than currency.), ProgTots does make TC costs a lot more forgiving, but also balancable. You can _buy_ a bulldog opp kit, But you'll need to meet the 23 TC cost. You want to start with the traitor implant because you're a PDA slop? That 4TC setback can be over-ruled with a few objectives. It opens up more room for error and discovery without immediately having harsh, possibly un-recoverable TC setbacks. You can afford losing an emag or two, or doing a funny smug-satchel gimmick of resource caches or removing proof from yourself, Without cutting tc that could be used for more... Game effecting/chaos reigning items like a station blackout or hacked AI modules. 
</details>

<details>
  <summary>Objectives objectives objectives</summary>
So, Let me kinda get this out on the get go. Progtots is very liberal with its objectives. _Yes, Thats a large problem people will complain about progtarts because a lot of the objectives are dumb,_ But having a choice between stealing the hand tele, killing Ian, Spraying a 3x3 tile art of the syndicate snake, Killing someone who likely isn't doing jack shit or are even having some fun time in dorm or on the interstation, or otherwise doing some other shit like breaking the ore silo, Is kinda... Good? Well, You are ultimately given the choice to do them or not, and you are ultimately rewarded for them in one way or another. Some of them are even fairly large in an inconspicuous way. Dont want to break into RD's locker for the protolathe board? You can steal the science protolathe for the bug objective and send it back through the mail system later, creating some instant drama that might get some attention. Yes, Some of them do have deductions if you fail them, meaning people will gamer them instead of using them to weave into their traitor story, Which is a valid complaint. But i dont.. Think having just 2 objectives that lead into an escape or suicide (i hate both of these because they're non-consequential. Either its a "make sure you survive and are on the shuttle, free" or "Die" goal thats likely to be done by doing your other objectives and playing the game anyways) end-goal is all that interesting. "Just opfor" just seems like a shrug-off to the large problem that is old-tot's flawed objective system 
</details>

<details>
  <summary>Personal opinions on potential improvements/suggestions</summary>
Honestly? I'd personally like to keep ProgTots. It has a functioning balancing act and progression guide to help lower over-escalation and keep more problematic people from jumping the gun at roundstart. I know we dont want to think about the old times that much, But roundstart Esword and hardsuits spinning down to medical bay to blow it up and cull all of security is something that, Yes, Is bannable and otherwise punishable, But a lot easier to stem with progtots, and would function a lot better with some adjustments.
But lets get down to just suggestions for the reverted system, because it seems its quite adamant that progtots is going to go out with the bathwater until some unfortunate soul has to make a case for it and be tasked with the grueling work to make it appealing to those that.. Just flat out might just not want the system at all even if "Improved".


- Honestly, We could use a fair amount of the progtot objectives. The 3x3 paintjob, protolathe destruction, smuggling, and (personally) the room bugging objectives are kinda fine, But might need some more tuning.
- Objective choice. Now, honestly? I'd probably suggest this in the form of togglable seperate antags, as some people want to be antags but probably aren't fans of sitting through the first 30 minutes of the round waiting for an admin to finish with opfors to address an ahelp asking for different objectives. 
-  Take a _GOOD_ long look at TC, whats being bought and how much more TC is being asked for in opfors. Again, Quite bluntly, Progtots was quite liberal with its usage of more TC. The roundstart implant was a viable option for even aggressive players because by the time they could actually use more aggressive tot items, they got that 4tc back and maybe had a delux ID or cham headset, And disposable/expendable items were more usable since you can always spend another small bit of TC Every so often if you did objectives.
- Maybe remove TC Kits. Like, I'm going to be just kinda blunt here, Some of them are really good for the 20TC cost at roundstart. Dark lord has a DEsword, Telekinesis __and__ item summons, Even if you disarm them, they can get their weapon back at the drop of a command. Robohand comes with ANTI DROP, A _VERY_ powerful option even to the stun baton, And the Mandalorian arms 14mm pistol which is quite a good option even if limited on its mags and magsize. And.. quite honestly? A lot of them _BEG_ for some sort of backstory or reason to be had for what they give, And were balanced on old-tots by being random in the same pool as... clown sniper, carp hunter, OOPS, ALL BOMBS, The laser ninja, and Mr Freeze.
</details>

**Comment by Zonespace27** [comment]
@Zenitheevee 

> I dont think they're _quite_ balanced to openly exist without the limit. These include; Choosable loadouts that contain more than its cost in TC worth of gear or items that are _otherwise_ not available in the base traitor system (Dark Lord, for the biggest and most outstanding example), Roundstart *quite aggressive* weapons (desword, bulldog, C20), Syndicate modsuits (Normal having a required 200, with the elite having a hefty 900 requirement) and even things like POWERSINKS being available behind a fair amount of time requirements.

These existed beforehand without the limit and worked fine, currently no limit

> But you'll need to meet the 23 TC cost. You want to start with the traitor implant because you're a PDA slop? That 4TC setback can be over-ruled with a few objectives. It opens up more room for error and discovery without immediately having harsh, possibly un-recoverable TC setbacks. You can afford losing an emag or two, or doing a funny smug-satchel gimmick of resource caches or removing proof from yourself, Without cutting tc that could be used for more... Game effecting/chaos reigning items like a station blackout or hacked AI modules.

I _will_ be looking into how much TC traitors start with once this gets full merged and settles for a bit, considering the power level of security.

> All your objective comments

I would be more than happy to hear any ideas you have that both encourage RP and don't encourage objective ping-pong.

> Honestly, We could use a fair amount of the progtot objectives. The 3x3 paintjob, protolathe destruction, smuggling, and (personally) the room bugging objectives are kinda fine, But might need some more tuning.

I was planning on porting some of the (better) objectives to regular tot once the stock TG system gets merged, help would be appreciated

> Objective choosing

Not a bad idea, letting tots choose their static objectives, or at least give them a small list to pick from

> Robohand bundle

You can't get that anymore (has been that way for a while)

> Re-randomming bundles

I'd be okay with that

if you have any other concerns you'd like addressed, ping me on discord

**Review by Gandalf2k15 — CHANGES_REQUESTED** [review]
remove all of the non-modular changes and disable it from rolling by changing the weight

**Comment by AzyreR** [comment]
Personally the main issue I had with progtot is that it had a bunch of interesting objectives but since they were both influenced by your job you most of the time did 2 of the interesting ones that stuck to your playstyle and then you were softlocked with, as an example: Kill x, behead x, plant a calling card on x, followed by steal this single item that is on a head at all times and will most likely end up with you having to kill atleast one or two people trying to obtain it and most likely killed/arrested cause you got no other options to take and no reroll.

Which is why personally the recommendation i saw of giving a list of static objectives people can choose from and set their timeline for the shift without having to depend entirely on RNG giving them what they want is a nice change, cause someone could easily take easy low threat objectives if they want a rather chill antag round or all the major ones like, steal the NAD and bug it, steal the R&D drive and bug it, steal the blackbox and destroy it ones and be a more loud or blatand conflict driven antag.

While also having the opfor option for more creative ones.

**Comment by Panicode** [comment]
This is a roleplay server. Why can't we just have both? Make you have a choice of what kinda antag you wanna be roundstart. Make them choose between Progtot OR your own ambition. This re-uses two advanced systems that were created for the supplement of roleplay and rounds we think fondly of post-tense.
A good middle ground would be this flow:
Roundstart > Roll antag > Choose between progtot and ambition
This combats the power of security, Progtot having the ability of earning TC, while ambitions give you extra TC (30... 35?) right then and there.
This gives both parties the choice of dynamic or freestyle antagonist roleplay, and incentivizes apt attitude.

Secondly about the auto-finalize ambition thing... Is it really that hard to punish players who abuse that mechanic? Banning/antagbanning or noting a player is considerably easier when it comes to weeding out bad antag players.

Someone can just put an ahelp or a report in for someone doing a bad ambition and it'll be solved, same thing as someone participating within LRP

I never comment on merges but this is my two cents.

**Comment by Zonespace27** [comment]
> This is a roleplay server. Why can't we just have both? Make you have a choice of what kinda antag you wanna be roundstart. Make them choose between Progtot OR your own ambition. This re-uses two advanced systems that were created for the supplement of roleplay and rounds we think fondly of post-tense. A good middle ground would be this flow: Roundstart > Roll antag > Choose between progtot and ambition This combats the power of security, Progtot having the ability of earning TC, while ambitions give you extra TC (30... 35?) right then and there. This gives both parties the choice of dynamic or freestyle antagonist roleplay, and incentivizes apt attitude.

Ambitions aren't coming back, for one, neither does this PR introduce such. The concern is that progtots are not encouraged to RP, and are instead encouraged to ping-pong between objectives as fast as possible.

> Secondly about the auto-finalize ambition thing... Is it really that hard to punish players who abuse that mechanic? Banning/antagbanning or noting a player is considerably easier when it comes to weeding out bad antag players.
> 
> Someone can just put an ahelp or a report in for someone doing a bad ambition and it'll be solved, same thing as someone participating within LRP

OPFOR is the replacement for ambitions and there are no plans of making it mandatory currently

**Comment by Zergspower** [comment]
To add more to what Zone states, from an admin perspective Progtots caused probably half of our tickets from many different angles. They were designed for rushing through and dont really have any RP motive behind them that adds to the round.

File: modular_skyrat/master_files/code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]: Single-letter variable bad.

File: modular_skyrat/master_files/code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm
Comment (GoldenAlpharex) [inline modular_skyrat/master_files/code/game/gamemodes/dynamic/dynamic_rulesets_midround.dm]: snake_case please.

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Here's a few things to fix.

**Review by GoldenAlpharex — APPROVED** [review]
Looks good to me!

**Review by GoldenAlpharex — CHANGES_REQUESTED** [review]
Actually, make sure you include this fix too! https://github.com/Skyrat-SS13/Skyrat-tg/pull/15780

**Review by GoldenAlpharex — APPROVED** [review]
I think we've run it long enough to confirm that it's working now.

**Pull request merged**
